### DISCUSSION TITLE
谈谈 MVX 中的 View · /mvx-view

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#257] [Announcements]
https://draveness.me/mvx-view

**Comment by SepCode** [comment]
- 看完文章才理解公司对View设计的思想。
    - 首先封装了Label，Button，TextEditor...等一系列显示控件，控件持有UIView子类视图
    - 然后封装了各种LinearLayoutPanel，FlowLayoutPanel，GridLayoutPanel等布局容器，容器持有UIScrollView视图
- 最重要的是但是：但是没有把控件自适应的计算从UIView中分离出来，使用UIView做自适应布局计算和它严重耦合在一起。导致主线程计算布局，严重影响性能和体验。
- 只需要把自适应计算逻辑从控件持有的UIView子类视图中抽出放到控件中计算，就可以实现子线程计算。但是需要自己维护自适应的计算逻辑。

**Comment by luoyibu** [comment]
不知道用autolayout的占比到底有多少，我们项目一直用得最原始的frame布局，楼主有数据不？

**Comment by shimachao** [comment]
我之前在win32上模仿Android的线性布局和绝对布局。发现要考虑的部分太多了。后来遇到一个输入法的问题没搞定，就放弃了。

**Comment by apbaca06** [comment]
文章有提到 CFRect，是否為筆誤？
應該為 CGRect?
謝謝
